File: Source/MKL/examples/vmlc/source/vsunpacki.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  float fA[VEC_LEN],fB1[VEC_LEN],fB2[VEC_LEN];

  int i=0,vec_len=VEC_LEN,incra=3;

  for(i=0;i<vec_len;i++) {
    fA[i]=(float)i+1;
    fB1[i]=0.0;
    fB2[i]=0.0;
  }

  vec_len=vec_len/incra+1;
  vsPackI(vec_len,fA,incra,fB1);
#if defined(MKL_VML_MIXED)
  vsUnpackI(vec_len,fB1,fB2,incra);
#elif defined(MKL_VML_LOWER)
  vsunpacki(&vec_len,fB1,fB2,&incra);
#elif defined(MKL_VML_UPPER)
  VSUNPACKI(&vec_len,fB1,fB2,&incra);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif

  printf("vsUnpackI test/example programm\n\n");
  printf("           Before packing             After packing          After Unpacking\n");
  printf("===============================================================================\n");
  for(i=0;i<VEC_LEN;i++) {
    printf("% 25.14f ",fA[i]);
    printf("% 25.14f ",fB1[i]);
    printf("% 25.14f",fB2[i]);
    printf("\n");
  }
}

File: Source/MKL/examples/vmlf/source/vsunpacki.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*4 sA(VEC_LEN)
	real*4 sB1(VEC_LEN)
	real*4 sB2(VEC_LEN)
	integer i,incra,len

	i=0
	incra=3

	do i=1,VEC_LEN
		sA(i)=i/1.0
		sB1(i)=0.0
		sB2(i)=0.0
	end do

	len=VEC_LEN/incra+1
	call VSPACKI(len,sA,incra,sB1)
	call VSUNPACKI(len,sB1,sB2,incra)

	print *,"vsUnpackI test/example programm"
	print *,""
	print *,""

	print *,"           Before packing             After packing          After Unpacking"
	print *,"==============================================================================="
	do i=1,VEC_LEN
		print 10,sA(i),sB1(i),sB2(i)
	end do

10	format (F25.13,F25.13,F25.13)

	end
